src/wampy.js

```diff
@@ -1350,144 +1350,115 @@ class Wampy {
-                        msg.push([]);
-                    }
-                    msg.push(argsDict);
-                }
-                self._send(msg);
-            },
-            invoke_result_handler = results => {
-                // WAMP SPEC: [YIELD, INVOCATION.Request|id, Options|dict, (Arguments|list,
-                // ArgumentsKw|dict)]
-                let msg = [WAMP_MSG_SPEC.YIELD, data[1], {}];
-
-                if (self._isPlainObject(results)) {
-
-                    if (self._isPlainObject(results.options)) {
-                        const options = results.options;
-
-                        // Check and handle Payload PassThru Mode
-                        // @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
-                        const pptScheme = results.options.ppt_scheme;
-
-                        if (pptScheme) {
-                            if (!this._checkPPTOptions('dealer', results.options)) {
-                                if (this._cache.opStatus.error && this._cache.opStatus.error instanceof Errors.PPTNotSupportedError) {
-                                    // This case should not happen at all, but for safety
-                                    this._hardClose('wamp.error.protocol_violation', 'Trying to send YIELD in PPT Mode, while Dealer didn\'t announce it');
-                                } else {
-                                    invoke_error_handler({
-                                        details : results.options,
-                                        error   : 'wamp.error.invalid_option',
-                                        argsList: [this._cache.opStatus.error.message],
-                                        argsDict: null
-                                    });
-                                }
-                                return;
-                            }
-
-                            options.ppt_scheme = pptScheme;
-
-                            if (results.options.ppt_serializer) {
-                                options.ppt_serializer = results.options.ppt_serializer;
-                            }
-                            if (results.options.ppt_cipher) {
-                                options.ppt_cipher = results.options.ppt_cipher;
-                            }
-                            if (results.options.ppt_keyid) {
-                                options.ppt_keyid = results.options.ppt_keyid;
-                            }
-                        }
-
-                        msg[2] = options;
-                    }
-                }
+        const handleInvocationError = ({ error, details, argsList, argsDict }) => {
+            self._send([
+                WAMP_MSG_SPEC.ERROR,
+                WAMP_MSG_SPEC.INVOCATION,
+                requestId,
+                details || {},
+                error || 'wamp.error.invocation_exception',
+                argsList || [],
```

Comment:
Well, this is not exactly the same as it was before.
The right way is:
* If there are no args — WAMP message should not hold even empty array
* If there are no kwargs — WAMP message should not hold even empty object.

These are 3 possible forms of ERROR message:
```
[ERROR, INVOCATION, INVOCATION.Request|id, Details|dict, Error|uri]
[ERROR, INVOCATION, INVOCATION.Request|id, Details|dict, Error|uri, Arguments|list]
[ERROR, INVOCATION, INVOCATION.Request|id, Details|dict, Error|uri, Arguments|list, ArgumentsKw|dict]
```